packages/data/src/redux-store/metadata/selectors.js

```diff
@@ -153,3 +153,33 @@ export function hasResolvingSelectors( state ) {
 		)
 	);
 }
+
+/**
+ * Retrieves the total number of selectors, grouped per status.
+ *
+ * @param {State} state Data state.
+ *
+ * @return {Object} Object, containing selector totals by status.
+ */
+export function countSelectorsByStatus( state ) {
+	const selectorsByStatus = {};
```

Comment:
Not sure how important that would be for consumers, but the selector will return a different value when called with the same state.

```js
// The assertion will fail.
expect( countSelectorsByStatus() ).toBe( countSelectorsByStatus() );
```

It could be mitigated by using a separate selector for statuses:

```js
const statuses = useSelect( ( select ) => select( 'store' ).countSelectorsByStatus(), [] );
```

Reply:
This is definitely a valid one, thanks for catching it! 🙌 

Since we still want a new object with every difference, as a solution I'm keeping the new object creation, but applying memoization in https://github.com/WordPress/gutenberg/pull/53767/commits/7a1076bef919213eba590ce0e50068fccbbddf12 so we'll be able to get a new object only when a different `state` is specified. In that commit, I've also added a couple of tests to confirm that this is working as expected.

Reply:
Thank you, @tyxla!